core/event.ml

```diff
@@ -38,14 +38,30 @@ module Location = struct
 end
 
 module Ok = struct
+  module Trace = struct
+    type t =
+      { thread : Thread.t
+      ; time : Time_ns.Span.t
+      ; trace_state_change : Trace_state_change.t option [@sexp.option]
+      ; kind : Kind.t option [@sexp.option]
+      ; src : Location.t
+      ; dst : Location.t
+      }
+    [@@deriving sexp]
+  end
+
+  module Power = struct
+    type t =
+      { thread : Thread.t
+      ; time : Time_ns.Span.t
+      ; freq : int
+      }
+    [@@deriving sexp]
+  end
+
   type t =
```

Comment:
If this type changes, we need to update the sexp-output version header too.

Reply:
https://github.com/janestreet/magic-trace/blob/782794dd1fa2df4f7f1b5d28ff4451f880318167/src/trace.ml#L120